dvc/repo/diff.py

```diff
@@ -0,0 +1,178 @@
+from __future__ import unicode_literals
+
+import os
+import humanize
+import inflect
+
+
+from dvc.scm.base import FileNotInCommitError
+
+
+def _extract_output(self, output):
+    lst = output.dir_cache
+    return {i["relpath"]: i["md5"] for i in lst}
+
+
+def _get_tree_changes(self, a_entries, b_entries):
+    result = {"del": 0, "ident": 0, "changes": 0, "new": 0, "moves": 0}
+    keys = set(a_entries.keys())
+    keys.update(b_entries.keys())
+    diff_size = 0
+    for key in keys:
+        if key in a_entries and key in b_entries:
+            if a_entries[key] == b_entries[key]:
+                result["ident"] += 1
+            else:
+                result["changes"] += 1
+                diff_size += os.path.getsize(
+                    self.cache.local.get(b_entries[key])
+                ) - os.path.getsize(self.cache.local.get(a_entries[key]))
```

Comment:
just out of curiosity, what happens if some files are not in cache? especially for the ref we are comparing to? how should users handle this in terms of workflow - we basically will need to give them an interface to pull a certain ref, and it could be an overkill to do that this way. It feels like we should actually ping remote to get the size, list of entries, etc. cc @efiop , what do you think?

Reply:
@shcheklein Good point. For now, I wouldn't ping remote and would simply put something like "?" or "unkown" into diff_size field.